frontend/javascripts/admin/user/user_list_view.js

```diff
@@ -107,48 +107,57 @@ class UserListView extends React.PureComponent<PropsWithRouter, State> {
     });
   }
 
-  activateUser = (selectedUser: APIUser, isActive: boolean = true): void => {
-    this.setState(prevState => {
-      const newUsers = prevState.users.map(user => {
-        if (selectedUser.id === user.id) {
-          const newUser = Object.assign({}, user, { isActive });
-          updateUser(newUser);
-          return newUser;
-        }
-        return user;
-      });
-
-      return {
-        users: newUsers,
-        selectedUserIds: [selectedUser.id],
-        isTeamRoleModalVisible: isActive,
-      };
+  activateUser = async (selectedUser: APIUser, isActive: boolean = true) => {
+    const newUserPromises = this.state.users.map(user => {
+      if (selectedUser.id === user.id) {
+        const newUser = Object.assign({}, user, { isActive });
+        return updateUser(newUser).then(
+          serverUser => Promise.resolve(serverUser),
+          () => Promise.reject(user),
+        );
```

Comment:
```suggestion
        return updateUser(newUser);
```

I think this should be equivalent for the success case and semantically "better" for the reject case. The argument to `reject` should be the "reason" for why the promise was rejected. It's recommended to use an error instance for that and I think that error instance is passed as the first parameter to the arrow-function (swallowed since you didn't list parameters for the error-handler). If the latter was changed, the entire `then` invocation becomes a no-op, if I'm not mistaken (hence, the above suggestion).